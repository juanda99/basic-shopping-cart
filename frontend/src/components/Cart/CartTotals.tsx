import React from 'react'
import { Product } from '../../store/modules/products/model'
import { VerticalBar } from './VerticalBar'
import styled from 'styled-components'

type Props = {
  cart: Product[]
  cartNumItems: number
}

export const CartTotals = ({ cart, cartNumItems }: Props) => {
  const cartPriceTotal = cart.reduce(
    (acc, product) => acc + product.customerPrice * product.quantity,
    0
  )

  return (
    <H2>
      Items: {cartNumItems} <VerticalBar /> Total Price: $
      {cartPriceTotal.toFixed(2)}
    </H2>
  )
}

const H2 = styled.h2`
  padding: 4px 0;
  font-size: 18px;
  border-bottom: 1px dashed black;
`

export default CartTotals
