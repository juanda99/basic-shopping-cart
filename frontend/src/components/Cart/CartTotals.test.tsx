import React from 'react'
import { shallow } from 'enzyme'
import CartTotals from './CartTotals'

const cart = [
  {
    id: 1,
    name: 'Soup',
    customerPrice: 199,
    cost: 186,
    inCart: false,
    quantity: 1,
  },
  {
    id: 2,
    name: 'Bread',
    customerPrice: 87,
    cost: 21,
    inCart: true,
    quantity: 2,
  },
]

describe('CartTotals', () => {
  it('Renders without errors', () => {
    shallow(<CartTotals cart={cart} cartNumItems={2} />)
  })
  it('Renders correctloy', () => {
    const wrapper = shallow(<CartTotals cart={cart} cartNumItems={2} />)
    expect(wrapper).toMatchSnapshot()
  })
})
