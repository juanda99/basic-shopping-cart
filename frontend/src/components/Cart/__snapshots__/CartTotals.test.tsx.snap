// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartTotals Renders correctloy 1`] = `ShallowWrapper {}`;
